From 7a37c06bbe8229c520a96e5a2b4f1b2c59eb028b Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 14 Mar 2018 19:20:32 +0800
Subject: [PATCH 3513/5242] MLK-17804-4: dts: change gpio pin for qxp mipi csi

commit  cbe2020ecc1e0588b2ccc868b6ed00ef39d394af from
https://source.codeaurora.org/external/imx/linux-imx.git

Replace mipi gpio pins with generic gpio pins.

Reviewed-by: sandor.yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit d6ce2512611822178b71bef3d5a566491b8fc146)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index e091fb2..0ea06f4 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -456,6 +456,14 @@
 				SC_P_MIPI_CSI0_GPIO0_01_MIPI_CSI0_GPIO0_IO01	0x00000021
 			>;
 		};
+
+		pinctrl_gpio3: gpio3grp{
+			fsl,pins = <
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO3_IO07		0xC0000041
+				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO3_IO08		0xC0000041
+			>;
+		};
+
 	};
 };
 
@@ -861,6 +869,11 @@
 	status = "okay";
 };
 
+&gpio3 {
+	pinctrl-name = "default";
+	pinctrl-0 = <&pinctrl_gpio3>;
+};
+
 &i2c0_csi0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -876,7 +889,7 @@
 		clock-names = "capture_mclk";
 		mclk = <27000000>;
 		mclk_source = <0>;
-		pwn-gpios = <&gpio0_mipi_csi0 0 GPIO_ACTIVE_HIGH>;
+		pwn-gpios = <&gpio3 7 GPIO_ACTIVE_HIGH>;
 		virtual-channel;
 		status = "okay";
 		port {
-- 
1.7.9.5

